lib/detect-testing-library-utils.ts

```diff
@@ -173,6 +174,69 @@ export function detectTestingLibraryUtils<
       return isNodeComingFromTestingLibrary(referenceNodeIdentifier);
     }
 
+    /**
+     * Determines whether a given node is a simulate event util related to
+     * Testing Library or not.
+     *
+     * In order to determine this, the node must match:
+     * - indicated simulate event name: fireEvent or userEvent
+     * - imported from valid Testing Library module (depends on Aggressive
+     * Reporting)
+     *
+     */
+    function isTestingLibrarySimulateEventUtil(
+      node: TSESTree.Identifier,
+      utilName: 'fireEvent' | 'userEvent'
+    ): boolean {
+      const simulateEventUtil = findImportedUtilSpecifier(utilName);
+      let simulateEventUtilName: string | undefined;
+
+      if (simulateEventUtil) {
+        simulateEventUtilName = ASTUtils.isIdentifier(simulateEventUtil)
+          ? simulateEventUtil.name
+          : simulateEventUtil.local.name;
+      } else if (isAggressiveModuleReportingEnabled()) {
+        simulateEventUtilName = utilName;
+      }
+
+      if (!simulateEventUtilName) {
+        return false;
+      }
+
+      const parentMemberExpression:
+        | TSESTree.MemberExpression
+        | undefined = isMemberExpression(node.parent) ? node.parent : undefined;
+
+      if (!parentMemberExpression) {
+        return false;
+      }
+
+      // make sure that given node it's not fireEvent/userEvent object itself
+      if (
+        [simulateEventUtilName, utilName].includes(node.name) ||
```

Comment:
I'm pretty sure this could be simplified by using `isTestingLibraryUtil` here but I don't want to start this refactor since it's working fine. I'll leave it for a future nice to have.